### PR TITLE
ADD: SearchService

### PR DESCRIPTION
**Pull request by wudong5**

### CODE REVIEW

**Review by zhaoyunxing92 — APPROVED** [review]

File: dubbo-admin-server/pkg/util/sync_utils.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package util
+
+import (
+	"admin/pkg/constant"
+	"admin/pkg/model"
+	"dubbo.apache.org/dubbo-go/v3/common"
+)
+
+func Url2Provider(id string, url *common.URL) *model.Provider {
```
Comment (justxuewei) [inline dubbo-admin-server/pkg/util/sync_utils.go]: Nit: Url2Provider -> URL2Provider

File: dubbo-admin-server/admin.go [hunk 2]
```diff
@@ -21,11 +21,24 @@ import (
...
+	"log"
+	"os"
+	"os/signal"
+	"syscall"
 )
 
 func main() {
 	config.LoadConfig()
-	services.StartSubscribe(config.RegistryCenter)
+	go services.StartSubscribe(config.RegistryCenter)
```
Comment (justxuewei) [inline dubbo-admin-server/admin.go:28]: It is better to put destroying subscription into a defer function, like below:

```go
go services.StartSubscribe(config.RegistryCenter)
defer func() {
    services.DestroySubscribe(config.RegistryCenter)
}()
```

File: dubbo-admin-server/pkg/model/entity.go [hunk 3]
```diff
@@ -0,0 +1,6 @@
+package model
+
+type Entity struct {
+	Id   string
```
Comment (justxuewei) [inline dubbo-admin-server/pkg/model/entity.go]: Nit: Id -> ID

File: dubbo-admin-server/pkg/model/provider.go [hunk 4]
```diff
@@ -0,0 +1,22 @@
+package model
+
+import "time"
+
+type Provider struct {
+	Entity
+	Service        string
+	Url            string
```
Comment (justxuewei) [inline dubbo-admin-server/pkg/model/provider.go]: Nit: Url -> URL

File: dubbo-admin-server/pkg/model/registry_source.go [hunk 5]
```diff
@@ -0,0 +1,11 @@
+package model
+
+type RegistrySource int
+
+const (
+	ALL RegistrySource = iota
```
Comment (justxuewei) [inline dubbo-admin-server/pkg/model/registry_source.go]: The constants should follow the Camel case. Thus the `ALL` should be `All`, and so on.

File: dubbo-admin-server/pkg/services/registry_service_sync.go [hunk 6]
```diff
@@ -73,14 +78,14 @@ func (adminNotifyListener) Notify(event *registry.ServiceEvent) {
 	if strings.EqualFold(constant.EmptyProtocol, serviceUrl.Protocol) {
 		if services, ok := cache.InterfaceRegistryCache.Load(category); ok {
 			if services != nil {
-				group := serviceUrl.GetParam(constant.GroupKey, "")
-				version := serviceUrl.GetParam(constant.VersionKey, "")
-				if constant.AnyValue == group && constant.AnyValue != version {
-					services.(*sync.Map).Delete(getServiceInterface(serviceUrl))
+				group := serviceUrl.Group()
```
Comment (justxuewei) [inline dubbo-admin-server/pkg/services/registry_service_sync.go]: Nit: serviceUrl -> serviceURL

File: dubbo-admin-server/pkg/services/registry_service_sync.go [hunk 7]
```diff
@@ -73,14 +78,14 @@ func (adminNotifyListener) Notify(event *registry.ServiceEvent) {
 	if strings.EqualFold(constant.EmptyProtocol, serviceUrl.Protocol) {
 		if services, ok := cache.InterfaceRegistryCache.Load(category); ok {
 			if services != nil {
```
Comment (justxuewei) [inline dubbo-admin-server/pkg/services/registry_service_sync.go:81]: The application will panic if the type assertion fails. Please add an inspection for the type assertion like:

```go
if map, ok := services.(*sync.Map); !ok {
    // handle errors of type assertion
}
```

Comment (wudong5) [reply dubbo-admin-server/pkg/services/registry_service_sync.go:81]: PTAL, thx

**Review by AlbumenJ — APPROVED** [review]

**Pull request merged**